### ISSUE TITLE
Docstring for Models

### ISSUE THREAD

**Issue by jkerfs** [blei-lab/edward#390]
By dynamically creating the RandomVariable classes, we lose the docstrings that describe the parameters of each distribution.  This is not a major issue, but it is certainly convenient to see information about the distribution in IDEs.  Below is an example of these docstrings from pymc3.
![pymc3-docstring](https://cloud.githubusercontent.com/assets/3375795/21627197/042901a6-d1cb-11e6-9b38-6397b86a033a.png).

I thought of three ways to resolve this:
1. Inherit the docstrings from tf.contrib.distributions.

This could be achieved by replacing the following (from edward.models.randomvariables.py):
```
    class _WrapperRandomVariable(RandomVariable, _candidate):
      def __init__(self, *args, **kwargs):
        RandomVariable.__init__(self, *args, **kwargs)

    _WrapperRandomVariable.__name__ = _name
    _globals[_name] = _WrapperRandomVariable
```

with 
```
_params = {"__doc__": _candidate.__doc__ }
_globals[_name] = type(_name, (RandomVariable, _candidate), _params)
```

The downside of this is that some of the docstrings contain code samples that reference the use of `tf.contrib.distributions`.  

2.  Generate docstring from the required arguments of the distribution:
```
_n_default_args = 1 + len(_canddiate.__init__.__defaults__)
_required_variables = _candidate.__init__.__code__.co_varnames[1:-_n_default_args]
_docstring = _name + " Distribution with parameters " + ", ".join(_required_variables)
params = {"__doc__": _docstring}
```

This would produce strings like `Beta Distribution with parameters a, b` and `Normal Distribution with parameters mu, sigma`.  Unfortunately, these docstrings are not very informative, and the python code to produce them becomes verbose and muddled.

3.  Statically define the class for each distribution and provide a short custom docstring

This would require the most work, but the docstrings would be useful, the parameters for each constructor could be named (enable autocomplete and suggestions in most IDEs), and it would play nice with static documentation compilers like doxygen.  Obviously the major negative is that new distributions added to tf.contrib.distributions would have to be manually added.

**Comment by dustinvtran** [comment]
Thanks for detailing the options! I agree that 3 would produce the most useful docstrings, at the cost of making it more difficult to maintain. Static classes also make it more difficult (although still possible) to support multiple tensorflow versions and not only the latest.

1 sounds like the best solution—perhaps not the end solution—but one worth having for now. I'd love to give you contributor credit and merge in a pull request that makes those simple changes. Other thoughts are appreciated.

**Comment by jkerfs** [comment]
Good points, I was leaning toward the first option myself.  I think the most important information for the docstrings is the names of the parameters and the formulas since many of the distributions have several common parameterizations.  Option number 1 provides this and is the easiest to maintain.

**Comment by dustinvtran** [comment]
fixed in #394
